cmd/admin-rpc-client.go

```diff
@@ -155,9 +180,51 @@ func sendServiceCmd(cps adminPeers, cmd serviceSignal) {
 		wg.Add(1)
 		go func(idx int) {
 			defer wg.Done()
-			errs[idx] = invokeServiceCmd(remotePeers[idx], cmd)
+			// we use idx+1 because remotePeers slice is 1 position shifted w.r.t cps
+			errs[idx+1] = invokeServiceCmd(remotePeers[idx], cmd)
 		}(i)
 	}
 	wg.Wait()
 	errs[0] = invokeServiceCmd(cps[0], cmd)
 }
+
+func fetchLocks(cps adminPeers, bucket, prefix string, relTime time.Duration) ([]VolumeLockInfo, error) {
+	// Used to aggregate volume lock information from all nodes.
+	allLocks := make([][]VolumeLockInfo, len(cps))
+	errs := make([]error, len(cps))
+	var wg sync.WaitGroup
+	remotePeers := cps[1:]
+	for i := range remotePeers {
+		wg.Add(1)
+		go func(idx int) {
+			defer wg.Done()
+			allLocks[idx+1], errs[idx] = cps[idx].cmdRunner.ListLocks(bucket, prefix, relTime)
+		}(i)
+	}
+	wg.Wait()
+	allLocks[0], errs[0] = cps[0].cmdRunner.ListLocks(bucket, prefix, relTime)
+
+	// Summarizing errors received for ListLocks RPC across all nodes.
+	errCount, err := reduceErrs(errs, []error{})
+	if errCount > (len(cps)/2+1) && err != nil {
```

Comment:
This means we return an error only when errCount == 4 in case of 4 nodes. It needs a little review here